docs/book/user-guide/llmops-guide/evaluation/generation.md

```diff
@@ -0,0 +1,370 @@
+Average helpfulness: 4.595
+Average relevance: 4.87
+Step e2e_evaluation_llm_judged has finished in 8m51s.
+Pipeline run has finished in 8m52s.
+```
+
+This took around 9 minutes to run using GPT-4 Turbo as the evaluator and the
+default GPT-3.5 as the LLM being evaluated.
+
+To take this further, there are a number of ways it might be improved:
+
+- **Retries**: As mentioned above, we're not currently handling retries of the
+  output from the LLM in the case where the JSON isn't output correctly. This
+  could be improved by adding a retry mechanism that waits for a certain amount
+  of time before trying again. (We could potentially use the
+  [`instructor`](https://github.com/jxnl/instructor) library to handle this
+  specifically.)
+- **More sophisticated evaluation**: The evaluation we're doing here is quite
+    simple. We're just asking for a score in four categories. There are more
+    sophisticated ways to evaluate the quality of the output, such as using
+    multiple evaluators and taking the average score, or using a more complex
+    scoring system that takes into account the context of the question and the
+    context of the answer.
+- **Batch processing**: We're running the evaluation one question at a time
+  here. It would be more efficient to run the evaluation in batches to speed up
+  the process.
+- **More data**: We're only using 50 samples here. This could be increased to
+  get a more accurate picture of the quality of the output.
+- **More LLMs**: We're only using GPT-4 Turbo here. It would be interesting to
+  see how other LLMs perform as evaluators.
+- **Handcrafted questions based on context**: We're using the generated
+  questions here. It would be interesting to see how the LLM performs when given
+  handcrafted questions that are based on the context of the question.
+- **Human in the loop**: The LLM actually provides qualitative feedback on the
+  output as well as the JSON scores. This data could be passed into an
+  annotation tool to get human feedback on the quality of the output. This would
+    be a more reliable way to evaluate the quality of the output and would offer
+    some insight into the kinds of mistakes the LLM is making.
+
+Most notably, the scores we're currently getting are pretty high, so it would
+make sense to pass in harder questions and be more specific in the judging
+criteria. This will give us more room to improve as it is sure that the system
+is not perfect.
+
+There are many ways to improve this evaluation, but this is a good starting
+point. Other frameworks do exist that can help with this kind of evaluation (for
+example, [`ragas`](https://github.com/explodinggradients/ragas),
+[`trulens`](https://www.trulens.org/),
+[DeepEval](https://docs.confident-ai.com/) and
+[UpTrain](https://github.com/uptrain-ai/uptrain)), but they're often quite
+complex to set up and use. When you're starting on a project, it can be better
+to start simple and then expand as needed.
```

Comment:
I don't really like this. Rather we should be saying that we can use these frameworks with ZenML depending on your use-case and familiarity with the underlying concepts